pkg/api/controllers/fileshare_test.go

```diff
@@ -88,6 +102,72 @@ var (
 	fakeFileShares = []*model.FileShareSpec{fakeFileShare}
 )
 
+func TestCreateFileShare(t *testing.T) {
+	var jsonStr = []byte(`{
+		"id": "d2975ebe-d82c-430f-b28e-f373746a71ca",
+		"name": "File_share",
+		"description": "fake File share",
+		"size": 1,
+		"profileId": "b3585ebe-c42c-120g-b28e-f373746a71ca",
+		"snapshotId": "b7602e18-771e-11e7-8f38-dbd6d291f4eg"
+	}`)
+
+	t.Run("Should return 202 if everything works well", func(t *testing.T) {
+		fileshare := model.FileShareSpec{BaseModel: &model.BaseModel{
+			Id:        "d2975ebe-d82c-430f-b28e-f373746a71ca",
+			CreatedAt: time.Now().Format(constants.TimeFormat),
+			UpdatedAt: time.Now().Format(constants.TimeFormat),
+		},
+			Name:             "File_share",
+			Description:      "fake File share",
+			Status:           "creating",
+			Size:             int64(1),
+			AvailabilityZone: "default",
+			ProfileId:        "b3585ebe-c42c-120g-b28e-f373746a71ca",
+			SnapshotId:       "b7602e18-771e-11e7-8f38-dbd6d291f4eg",
+		}
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetDefaultProfileFileShare", c.NewAdminContext()).Return(&SampleProfiles[0], nil)
+		mockClient.On("GetFileShareSnapshot", c.NewAdminContext(), fileshare.SnapshotId).Return(&SampleFileShareSnapshots[0], nil)
+		mockClient.On("GetFileShare", c.NewAdminContext(), SampleFileShareSnapshots[0].FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("GetProfile", c.NewAdminContext(), fileshare.ProfileId).Return(&SampleFileShareProfiles[0], nil)
+		mockClient.On("CreateFileShare", c.NewAdminContext(), &fileshare).Return(&SampleFileShares[0], nil)
+		db.C = mockClient
+
+		r, _ := http.NewRequest("POST", "/v1beta/file/shares", bytes.NewBuffer(jsonStr))
```

Comment:
The URI is usullay /v1beta/{tenantId}/file/shares ..But we are not taking tenantId in consideration..Isn't this will make coverage but not actual Unit test?

Reply:
tenantId is used if it is AUTH_ENABLED , it should be fine for unit without that.

Reply:
Ok